ambari-common/src/main/python/ambari_commons/credential_store_helper.py

```diff
@@ -19,27 +19,66 @@
 """
 
 import os
+import re
 
-from resource_management.core.resources.system import File
+from resource_management.core.resources.system import File, Execute
 from resource_management.core.shell import checked_call
 from resource_management.core.source import DownloadSource
+from resource_management.core.utils import PasswordString
 
 credential_util_cmd = 'org.apache.ambari.server.credentialapi.CredentialUtil'
 credential_util_jar = 'CredentialUtil.jar'
 
-def get_password_from_credential_store(alias, provider_path, cs_lib_path, java_home, jdk_location):
+def removeloglines(lines):
+    regex = re.compile(r'^(([0-1][0-9])|([2][0-3])):([0-5][0-9])(:[0-5][0-9])[,]\d{1,3}')
+    cleanlines = filter(regex.search, lines)
+    return(cleanlines)
+
+def downloadjar(cs_lib_path, jdk_location):
     # Try to download CredentialUtil.jar from ambari-server resources
-    credential_util_dir = cs_lib_path.split('*')[0] # Remove the trailing '*'
+    credential_util_dir = cs_lib_path.split('*')[0].split(":")[-1:][0] # Remove the trailing '*' and get the last directory if an entire path is passed
```

Comment:
What's the difference between .split(":")[-1:][0] and .split(":")[-1] ?

Reply:
This is to account for the fact when the class path is multiple directories. Some services like Atlas require multiple directories to run the jar (webapps META-INF and the conf directory). 

The change here specifically will find the last folder in the class path and download the CredentialAPI.jar to it.

Reply:
But if the intention was to find the last one then you can use [-1].

```python
>>> a = [1,2,3]
>>> a[-1:][0]
3
>>> a[-1]
3
```

Reply:
You are right, This is more efficient. Would you like me to change it ?

Reply:
yes please